hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteObservableP.java

```diff
@@ -0,0 +1,132 @@
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.core.Watermark;
+import com.hazelcast.jet.impl.observer.ObservableUtil;
+import com.hazelcast.jet.impl.util.Util;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.ringbuffer.OverflowPolicy;
+import com.hazelcast.ringbuffer.Ringbuffer;
+import com.hazelcast.ringbuffer.impl.RingbufferProxy;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+
+public final class WriteObservableP<T> implements Processor {
+
+    private static final int ASYNC_OPS_LIMIT = 1;
+    private static final int MAX_BATCH_SIZE = RingbufferProxy.MAX_BATCH_SIZE;
+
+    private final String ringbufferName;
+    private final List<T> batch = new ArrayList<>(MAX_BATCH_SIZE);
+    private final AtomicInteger pendingWrites = new AtomicInteger(0);
+
+    private Ringbuffer<Object> ringbuffer;
+    private ILogger logger;
+
+
+    private WriteObservableP(String ringbufferName) {
+        this.ringbufferName = ringbufferName;
+    }
+
+    @Override
+    public void init(@Nonnull Outbox outbox, @Nonnull Context context) {
+        HazelcastInstance instance = context.jetInstance().getHazelcastInstance();
+        this.logger = context.logger();
+        this.ringbuffer = instance.getRingbuffer(ObservableUtil.getRingbufferName(ringbufferName));
+    }
+
+    @Override
+    public void process(int ordinal, @Nonnull Inbox inbox) {
+        if (batch.isEmpty()) {
+            inbox.drainTo(batch, MAX_BATCH_SIZE);
+        }
+        tryFlush();
+    }
+
+    @Override
+    public boolean tryProcess() {
+        return tryFlush();
+    }
+
+    private boolean tryFlush() {
+        if (batch.isEmpty()) {
+            return true;
+        }
+        if (Util.tryIncrement(pendingWrites, 1, ASYNC_OPS_LIMIT)) {
+            ringbuffer.addAllAsync(batch, OverflowPolicy.OVERWRITE)
+                    .whenComplete(this::onFlushComplete);
+            batch.clear();
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private void onFlushComplete(Long lastSeq, Throwable throwable) {
+        if (throwable != null) {
+            logger.warning("Failed publishing into observable '" + ringbufferName + "'", throwable);
+            //TODO (PR-1729): extract observable name from ringbuffer name
+        }
+        pendingWrites.decrementAndGet();
+    }
+
+    @Override
+    public boolean tryProcessWatermark(@Nonnull Watermark watermark) {
+        // we're a sink, no need to forward the watermarks
+        return true;
+    }
+
+    @Override
+    public boolean complete() {
+        return tryFlush() && pendingWrites.get() <= 0;
+    }
+
+    public static <T> ProcessorSupplier supplier(String name) {
+        return new Supplier<T>(name);
```

Comment:
We can return a `SupplierEx<Processor>`, there's nothing in the Supplier that necessitates it.